File: lab 3/Domino Solver/Domino Solver/MathFunctions.cs
using System;
using System.Collections.Generic;
using System.Text;

class MathFunctions
{
    public float Norm(List<float> values, float pow)
    {
        float val = 0;
        foreach (float n in values)
            val += MathF.Pow(n, pow);
        return MathF.Pow(val, 1 / pow);
    }
}

File: lab 3 Domino Maze Solver/Domino Solver Cpp/Domino Solver Cpp/math_functions.cpp
#include "math_functions.h"

template<typename T, typename C>
T Math_Functions::norm(std::list<T>* vars, C p)
{
	T val = 0;
	for (T num : vars)
		val += pow(num, p);
	return pow(val, 1 / p);
}